publish/publish-type.proj

```diff
@@ -0,0 +1,114 @@
+    <Error Condition="'$(PublishBlobFeedKey)' == ''" Text="Missing required property 'PublishBlobFeedKey'" />
+
+    <PropertyGroup>
+      <!-- map the properties to the values expexed by the Feeds.targets file -->
+      <ExpectedFeedUrl>$(PublishBlobFeedUrl)</ExpectedFeedUrl>
+      <AccountKey>$(PublishBlobFeedKey)</AccountKey>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="DownloadFilesFromContainer">
+    <!-- TODO: We should also think about pulling down the checksums or recomputing them -->
+    <PropertyGroup>
+      <DownloadDirectory Condition="'$(DownloadDirectory)' == ''">$(BinDir)PackageDownload/</DownloadDirectory>
+      <PublishDirectory Condition="'$(PublishDirectory)' == ''">$(BinDir)ForPublishing/</PublishDirectory>
+      <!-- if we're not signing packages, publish directly from the download directory, as we won't be
+	       copying them to the indexed directory -->
+      <PublishDirectory Condition="'$(WindowsSdkDir)' == '' OR '$(SignType)' != 'real'">$(DownloadDirectory)</PublishDirectory>
+    </PropertyGroup>
+    <MakeDir Directories="$(DownloadDirectory)" Condition="!Exists('$(DownloadDirectory)')" />
+
+    <DownloadFromAzure AccountName="$(AzureAccountName)"
+                       AccountKey="$(AzureAccessToken)"
+                       ContainerName="$(ContainerName)"
+                       BlobNamePrefix="$(BinariesRelativePath)"
+                       DownloadDirectory="$(DownloadDirectory)" />
+
+    <ItemGroup>
+      <!-- ShippedNugetPackage item is used for updating versions repo -->
+      <ShippedNuGetPackage Include="$(DownloadDirectory)**\*.nupkg" Exclude="$(DownloadDirectory)**\*.symbols.nupkg" />
+
+      <!-- Setup item groups for PublishPackagesToBlobFeed and PublishFilesToBlobFeed -->
+      <PackagesToPublish Include="$(DownloadDirectory)**\*.nupkg" />
+      <FilesToPublish Include="$(DownloadDirectory)**\*.*" Exclude="$(PackagesToPublish)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="FindDownloadedPackagesForSigning">
+    <ItemGroup>
+      <!-- Find the downloaded symbol packages -->
+      <SymbolPackagesToEmbedIndex Include="$(DownloadDirectory)**\*.symbols.nupkg" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <UnsignedSymbolsDirectory Condition="'$(UnsignedSymbolsDirectory)' == ''">$(BinDir)UnsignedSymbolsDirectory/</UnsignedSymbolsDirectory>
+      <!-- Glob matching packages that we want to embed symbol signatures in. Used in BuildTools. -->
+      <SymbolPackagesToPublishGlob Condition="'$(SymbolPackagesToPublishGlob)' == ''">$(DownloadDirectory)**\*.symbols.nupkg</SymbolPackagesToPublishGlob>
+    </PropertyGroup>
+
+    <!-- Move all the symbol packages to an unsigned directory -->
+    <MakeDir Directories="$(UnsignedSymbolsDirectory)" />
+    <Move SourceFiles="@(SymbolPackagesToEmbedIndex)" DestinationFolder="$(UnsignedSymbolsDirectory)" />
+  </Target>
+
+  <Target Name="SignSymbolPackages"
```

Comment:
This appears to be duplicated with `publish.proj`.  Is there a way to factor it so this isn't duplicated?

Reply:
Maybe a good question is - why are these split into 2 separate projects?  Why not have a single project that takes the `PublishType` parameter and does the necessary operation?  This approach seems like it would simplify the `Core-Setup-Publish.json` file as well.

---
In reply to: [150610236](https://github.com/dotnet/core-setup/pull/3404#discussion_r150610236) [](ancestors = 150610236)

Reply:
I agree we need to overhaul this but my plans right now is to slowly replace the existing project with this new configurable one. I just didn't want to break the existing scenarios right now so I'm doing a targeted move. If for some reason we cannot fully replace then I will look at better ways to share.

Reply:
Instead of building up debt, why not make the existing one configurable as we go?

One approach could be to set properties at the top of publish.proj based on the PublishType property.

```xml
<ShouldDoXX Condition="$(PublishType) == 'nopublishtype' or $(PublishType.Contains('XX')">true</ShouldDoXX>
```

Reply:
That is the refactoring I'm referring to and I plan on continuing that work but this change is a step to enable us to start producing orchestrated builds.